compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -299,12 +299,12 @@ static FailureOr<Value> gpuRequireMemSpaceAllocationFn(OpBuilder &builder,
                                                        ValueRange dynamicSizes,
                                                        unsigned alignment) {
   Attribute memorySpace = memRefType.getMemorySpace();
-  // Bail out if the memref type specifies a nonnull memory space that is not
-  // #gpu.address_space.
+  // Default to private space if the memref type specifies a nonnull memory
```

Comment:
You probably need a test, and I am not entirely sure why this is taking something that has a memory space and setting it to null. It seems like `failure` is the best course here.